src/eiffel_graphql_api/graphql/db/database.py

```diff
@@ -42,20 +58,98 @@ def connect(mock):
                                              urllib.parse.quote(password),
                                              host, port)
     else:
-        url = "mongodb://{}:{}".format(host, port),
+        url = "mongodb://{}:{}".format(host, port)
     CLIENT = mongo_client(url, replicaset=replicaset)
     DATABASE = CLIENT[database_name]
+    # 'server_info' will create a connection against the MongoDB effectively testing
+    # the connection for us. This means that the 'connect' method will block until
+    # a connection can be established or 'ServerSelectionTimeoutError' is raised.
+    LOGGER.info("%r", CLIENT.server_info())
+    LOGGER.info("Connected.")
+
+
+def retry(fixture, timeout, *args, **kwargs):
+    """Retry call to fixture for 'timeout' seconds.
+
+    :param fixture: Fixture to call upon.
+    :type fixture: :obj:`func`
+    :param timeout: How long to wait for fixture to succeed. In seconds.
+    :type timeout: float
+    :param args: Positional argument to pass onto fixture.
+    :type args: tuple
+    :param kwargs: Keyword arguments to pass onto fixture.
+    :type kwargs: dict
+    """
+    end = time.time() + timeout
+    while time.time() < end:
+        try:
+            fixture(*args, **kwargs)
+            break
+        except Exception as exception:  # pylint:disable=broad-except
+            LOGGER.warning("%r", exception)
+            LOGGER.warning("Retrying %r", fixture)
+            time.sleep(0.1)
```

Comment:
This is a really short sleep that'll flood the logs with 20 messages/second (assuming failures are fast, e.g. ECONNREFUSED). A backoff is pretty standard.

Reply:
Removed retry in favor of MongoDB handling retries